CONTRIBUTING.md

```diff
@@ -13,3 +13,40 @@ in other words it allows to separate the actual name of the variable and the nam
 
 In the case of a pseudo-local variable (a field used briefly to pass around a local variable of a method between 2 injections of said method),
 the field should be named with the namespace first, then the name of the injected method, and finally the name of the local (`quilt$injectedMethod$localName`).
+
+## Licensing
+
+Quilt Standard Libraries are licensed under [Apache 2.0][LICENSE], which means your contributions should not introduce code of incompatible licensing.
+
+### License Headers
+
+Every Java source file in QSL has a license header, it allows keeping track of creation date,
+and last modification for the copyright notice of each file individually.
+
+Two gradle tasks are dedicated to them:
+ - `:checkLicenses` will check the presence and validity of the license header in each Java source file;
+ - `:applyLicenses` will automatically apply to Java source file the correct license header,
+    it can be used to append the header if it's missing, to update dates, and to update in the whole code base easily
+    every license headers if necessary.
```

Comment:
```suggestion
 - `:applyLicenses` will automatically apply the correct license header to every Java source file;
    it can be used to add missing headers and update copyright dates across the whole code base.
```